client/allocdir/alloc_dir.go

```diff
@@ -217,6 +231,57 @@ func (d *AllocDir) MountSharedDir(task string) error {
 	return nil
 }
 
+// List returns the list of files at a path relative to the alloc dir
+func (d *AllocDir) List(path string) ([]*AllocFileInfo, error) {
+	p := filepath.Join(d.AllocDir, path)
+	finfos, err := ioutil.ReadDir(p)
+	if err != nil {
+		return []*AllocFileInfo{}, err
+	}
+	files := make([]*AllocFileInfo, len(finfos))
+	for idx, info := range finfos {
+		files[idx] = &AllocFileInfo{
+			Name:  info.Name(),
+			IsDir: info.IsDir(),
+			Size:  info.Size(),
+		}
+	}
+	return files, err
+}
+
+// Stat returns information about the file at path relative to the alloc dir
+func (d *AllocDir) Stat(path string) (*AllocFileInfo, error) {
+	p := filepath.Join(d.AllocDir, path)
+	info, err := os.Stat(p)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AllocFileInfo{
+		Size:  info.Size(),
+		Name:  info.Name(),
+		IsDir: info.IsDir(),
+	}, nil
+}
+
+// ReadAt returns a reader  for a file at the path relative to the alloc dir
+//which will read a chunk of bytes at a particular offset
+func (d *AllocDir) ReadAt(path string, offset int64, limit int64) (io.ReadCloser, error) {
+	p := filepath.Join(d.AllocDir, path)
+	f, err := os.Open(p)
+	if err != nil {
+		return nil, err
+	}
+	return &FileReadCloser{Reader: io.LimitReader(f, limit), Closer: f}, nil
+}
+
+// FileReadCloser wraps a LimitReader so that a file is closed once it has been
```

Comment:
I would rename to ReadCloserWrapper and change the comment from that it wraps a limit reader b/c all it cares about is the reader aspect